### PR TITLE
[Playwright] PEPPER-877 Playwright tests for DSM participant list download

### PR DESCRIPTION
**Pull request by aweng98**
[PEPPER-877](https://broadworkbench.atlassian.net/browse/PEPPER-877)

New tests:

- `download-all-data.spec.ts`
- `download-subset-data.spec.ts`

Modified some functions to work with new UI.

[PEPPER-877]: https://broadworkbench.atlassian.net/browse/PEPPER-877?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: playwright-e2e/tests/osteosarcoma/adult-and-child-enrollment.spec.ts [hunk 1]
```diff
@@ -109,7 +109,12 @@ test('Osteo enroll self and kid together @osteo', async ({ page }) => {
   await page.getByText('By completing this information, you are agreeing to allow us to contact these ph').click();
   await page.getByText('I have already read and signed the informed consent document for this study, whi').click();
   await page.getByLabel('Full Name').click();
-  await page.getByRole('combobox', { name: 'Full Name' }).fill('Playwright McTestsAlot');
+
+  const responsePromise = waitForResponse(page, { uri: '/answers'});
+  await Promise.all([
+    responsePromise,
+    page.getByRole('combobox', { name: 'Full Name' }).fill('Playwright McTestsAlot')
+  ]);
```
Comment (aweng98) [inline playwright-e2e/tests/osteosarcoma/adult-and-child-enrollment.spec.ts:117]: Prevents clicking `Submit` button too early.

File: playwright-e2e/authentication/auth-dsm.ts [hunk 2]
```diff
@@ -15,7 +16,10 @@ export async function login(page: Page, opts: { email?: string; password?: strin
   if (DSM_BASE_URL == null) {
     throw Error('Invalid parameter: DSM base URL is undefined or null.');
   }
+
   await page.goto(DSM_BASE_URL, { waitUntil: 'load' });
   await fillInEmailPassword(page, { email, password, waitForNavigation: false });
+
+  await page.locator('.auth0-loading').waitFor({ state: 'hidden' });
```
Comment (aweng98) [inline playwright-e2e/authentication/auth-dsm.ts:23]: Prevents checking of spinner too early.

File: playwright-e2e/tests/dsm/participant-list/download-all-data.spec.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      await participantListPage.selectAll();
+
+      // Export as human-readable, Excel and include all completion of an activity
+      const download = await participantListPage.downloadParticipantList();
+      assertParticipantListDownloadFileName(download, study);
+
+      const downloadedFile = await download.path();
+      expect(downloadedFile).toBeTruthy();
+
+      // TODO unzip and check xlsx contents
```
Comment (aweng98) [inline playwright-e2e/tests/dsm/participant-list/download-all-data.spec.ts:27]: Download file is a `.zip` file which contains the `.xlsx` file. In order to read the zip file, I need to install a new third-party lib.

File: playwright-e2e/dsm/pages/participant-list-page.ts [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
 import { StudyNavEnum } from 'dsm/component/navigation/enums/studyNav-enum';
 import { Navigation } from 'dsm/component/navigation/navigation';
 import { WelcomePage } from 'dsm/pages/welcome-page';
+import Checkbox from 'dss/component/checkbox';
 import { waitForNoSpinner, waitForResponse } from 'utils/test-utils';
 import { Filters } from 'dsm/component/filters/filters';
 import { ParticipantListTable } from 'dsm/component/tables/participant-list-table';
 
+export enum FileFormatEnum {
+  XLSX = 'Excel (.xlsx)',
```
Comment (GiCharkviani) [inline playwright-e2e/dsm/pages/participant-list-page.ts]: [OPTIONAL]
What do you think about moving enums to a separate file?

Comment (aweng98) [reply playwright-e2e/dsm/pages/participant-list-page.ts]: sure. I don't have any preference. But we should find a new way to better organize enums for easy finding.

**Review by GiCharkviani — APPROVED** [review]
LGTM

**Pull request merged**